web/src/test/java/org/springframework/security/web/server/csrf/CsrfWebFilterTests.java

```diff
@@ -284,4 +305,12 @@ String ok() {
 			return "ok";
 		}
 	}
+
+	static class MockNonStandardHttpRequest {
```

Comment:
Is there a need to externalize this or could you just directly use `MockServerHttpRequest.method` directly?

Reply:
The idea was to build MockServerWebExchange from NonStandard HTTP Request (i.e: "PURGE", "REVOKE" etc.), by using user defined string which is not a valid HTTP Method.

Now mocking without externalizing meaning I have to use get, post or other defined builders in MockServerHttpRequest (MockServerHttpRequest itself not extensible) or even if I try to do in different way, most likely i would end up writing redundant code.

Please share your thought.

Reply:
It looks like it is only being used once, so I think it can safely be consolidated without having redundant code.